reference/7.1/Microsoft.PowerShell.Core/About/about_Parameters.md

```diff
@@ -21,11 +21,25 @@ parameters follow the command name and have the following form:
 -<parameter_name> <parameter_value>
 -<parameter_name>:<parameter_value>
 ```
-
-The name of the parameter is preceded by a hyphen (-), which signals to PowerShell that the word
-following the hyphen is a parameter name. The parameter name and value can be separated by a space
-or a colon character. Some parameters do not require or accept a parameter value. Other parameters
+<!--
+01234567890123456789012345678901234567890123456789012345678901234567890123456789
+-->
+The name of the parameter is preceded by a hyphen (`-`), which signals to
+PowerShell that the word following the hyphen is a parameter name.
+This effect can be prevented by escaping the hyphen (``Write-Output `-I``)
+or specifying the string as a parameter value (`Write-Output -InputObject:-I`).
+The special parameter `--` switches off recognising parameters for the rest
+of the command (`Write-Output -- -I`).
+The parameter name and value can be separated by a space or a colon character.
+Some parameters do not require or accept a parameter value. Other parameters
 require a value, but do not require the parameter name in the command.
+Named parameters passed to a command that does not declare them as parameters
```

Comment:
I'm not sure about this. While it does work for Write-Output, if you try and do somethin similar with Get-Item, You get an Error saying  A parameter cannot be found.... 

The -Path param is a positional param for Get-Item but there error returned isn't about the path being able to be found, the expected error for that. This behavior is much more common in my experience for the parameters. So is there a way to specify it more or is it something that shouldn't be brounght up here as it's uncommon behavior?

Reply:
We should not be using `Write-Object` for examples on how parameters work. It is a special case.

The **InputObject** parameter has the attribute `ValueFromRemainingArguments = true`. That means that any string that does not match a defined parameter is treated as a value that is assigned to **InputObject**. If we document anything about `Write-Object` in this article, we need to document the resulting behavior of `ValueFromRemainingArguments`. It should be a separate section of the document with its own examples.

Reply:
This behaviour is not that uncommon, given that any simple function and any external application will get named parameters that way.  `ValueFromRemainingArguments` is a special provision to relax requirements of strict cmdlets.  I am more worried about user functions getting unexpected parameters that way than about users getting errors from passing unexpected parameters to strict cmdlets; the latter is expected, the former—not so.  In other words, if a parameter specification is used to set a parameter value within a function, an unexpected parameter is expected to do nothing or error out.  It is not expected to be passed as a string, which is why I think it should be documented.

Reply:
If my testing is correct, this is pretty uncommon. Less than 5 percent of cmdlets that even have the `ValueFromRemainingArguments` parameter attribute actually accept it as a string or string array this way. This includes custom functions, Imported modules and all standard modules.  18 commands work and over 400 don't. 

Even in a simple function (nothing defining it as an advanced function. No cmdletBinding and not parameter attributes), while the bad parameter doesn't throw an error, it also doesn't seem to be accessible in the function. Not in the input variable or in any form. If you add a `Position` parameter attribute (making it act as an advanced function), the defualt behavior seems be to throw the "A parameter cannot be found... " error.

I don't see how this adds to this article for someone trying to understand the normal behavior of parameters. If there is something in about_Advanced_Functions or something similar that can be added to clarify the behavior that you're talking about, (which, I agree, does exist, just not to the extent of "normal" behavior) I'm open to that. But I don't see it adding value in this article.

Reply:
This is not about cmdlets, it is about user functions and applications.  There are infinitely many more of those than of cmdlets.  The extraneous arguments are perfectly accessible: 
```powershell
& { $ARGS } -A:A
CMD /CECHO -A:A
```
If PowerShell is a shell, its primary reason for existence is to invoke applications.  PowerShell functions are not cmdlets by default.  Thus this is the normal behaviour while cmdlet behaviour is special behaviour.

Reply:
I think if we're documenting things in enough detail that the differences in parameter binding for different command types are appearing, then we need to acknowledge that different kinds of commands bind parameters differently.

There are commonalities that we can introduce initially as introductory concepts, but cmdlets, functions/scriptblocks and native commands all have their own processors and parameter binders, so different concepts apply.

The way a command handles different parameters simply depends on what kind of command it is.

I think trying to conflate all kinds of commands together in a single description will ultimately just prove confusing, since `named parameters passed to a command that does not declare them as parameters become positional parameters` may be true for e.g. native commands, but `If a parameter name is abbreviated... PowerShell picks the parameter that begins with that prefix` is not true of native commands at all. That may seem clear to us, but I suspect such a line could create such an expectation in users.

My recommendation is to give a broad introduction of parameters and how they are generally used/passed to commands, and then give more detailed explanations in separate specialised paragraphs pertaining to particular kinds of commands.